### PR TITLE
Fix InstanceMetadataOptions defaults

### PR DESCRIPTION
**Pull request by muraee**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fixes `InstanceMetadataOptions` defaults not being set for AWSMachines

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4037#discussion_r1129064749

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Fix InstanceMetadataOptions defaults not being set for AWSMachines
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
@muraee could you please fix the test?

**Comment by Ankitasw** [comment]
Same tests is failing again and again due to new change, I think we need to debug the problem in those tests.

**Comment by Ankitasw** [comment]
Still failing with same error 😢

**Comment by Ankitasw** [comment]
> We need to fallback to using aws API directly instead and therefore we should populate [aws-secret](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/08efc3f5489af3557fb359c9ea057d2f7ee0f69d/test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/addons/csi/data/aws-ebs-csi-external.yaml#L1-L8) resource in the test.
I suppose we could simply use the ${AWS_ACCESS_KEY_ID} and ${AWS_SECRET_ACCESS_KEY} env vars to populate the secret, right?

Yes it should be ok.

> We probably want to bump the ebs driver version as well.

This should be in scope of this PR as this PR is itself blocking it.

**Comment by muraee** [comment]
@Ankitasw the aws-ebs-csi related tests passed, but now one of the CAPI tests is failing :(
I couldn't find why, could you please take a look?

**Comment by Ankitasw** [comment]
@richardcase @Skarlso need a little help here.
I fail to understand why and how the mhc remediation is affected because of introducing instance metadata options. The machine is not going to unhealthy state even if the condition is patched in the test.

**Comment by Ankitasw** [comment]
/lgtm
/approve
/hold

until test passes.
@muraee feel free to unhold after test passes.

File: test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/addons/csi/data/aws-ebs-csi-external.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ metadata:
   name: aws-secret
   namespace: kube-system
 stringData:
-  key_id: ""
```
Comment (wyike) [inline test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/addons/csi/data/aws-ebs-csi-external.yaml:7]: Use explict key and access key in csi yaml should not be the best practice.
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4247 This support ideally will allow we zero the aws credentials in csi yaml.

Comment (Ankitasw) [reply test/e2e/data/infrastructure-aws/withoutclusterclass/kustomize_sources/addons/csi/data/aws-ebs-csi-external.yaml:7]: you could make a change in a separate PR

**Pull request merged**